### ISSUE TITLE
Support for scalatest 3.0.0-RC1

### ISSUE THREAD

**Issue by zekefast** [paulbutcher/ScalaMock#139]
# Overview

`scalatest` of version `3.0.0-RC1` is just [out](https://github.com/scalatest/scalatest/releases/tag/release-3.0.0-RC1-for-scala-2.11-and-2.10).
It would be great to have a compatible build of `scalamock-scalatest-support`.
# Actual Result

Right now tests which have passed with `scalatest` `v3.0.0-m15` are failing with `3.0.0-RC1` with follow error.

```
[error] missing or invalid dependency detected while loading class file 'AbstractMockFactory.class'.
[error] Could not access type NoArgTest in trait org.scalatest.Suite,
[error] because it (or its dependencies) are missing. Check your build definition for
[error] missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
[error] A full rebuild may help if 'AbstractMockFactory.class' was compiled against an incompatible version of org.scalatest.Suite.
[error] one error found
```
# Environment
- `scala`: `2.11.7`
- `scalatest`: `3.0.0-RC1`
- `scalamock`: `3.2.2`

**Comment by ngbinh** [comment]
Yes, this is the issue we face too.

**Comment by mysticfall** [comment]
Same here. It's especially crucial for those who use Scala.js, since Scalatest began supporting it since 3.0.0.

**Comment by fommil** [comment]
me too

**Comment by fommil** [comment]
as a workaround, we could add https://github.com/paulbutcher/ScalaMock/tree/master/frameworks/scalatest/src/main/scala/org/scalamock/scalatest as a source dependency, does anybody know what changes would be needed?

**Comment by fommil** [comment]
```
[error] /home/fommil/Projects/ensime-server/testutil/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:31: not found: type NoArgTest
[error]   abstract override def withFixture(test: NoArgTest): Outcome = {
```

**Comment by fommil** [comment]
step 1: change `Suite` to `TestSuite`

**Comment by fommil** [comment]
step 2: change the `this: TestSuite =>` to an `extends`

**Comment by damirv** [comment]
Did anyone find a way to use scalamock with the latest releases of scalatest 3 ?

**Comment by damirv** [comment]
Scalatest 3.0.0 is now released, in the mean time, did anyone manage to get it working?

**Comment by fommil** [comment]
So my 2 step guide wasn't any use?

**Comment by ShaneDelmore** [comment]
So you don't feel like you wasted your time, your 2 step guide was super useful, thanks.

**Comment by damirv** [comment]
@fommil It worked! I have just submitted a PR for scalatest 3.0.0: https://github.com/paulbutcher/ScalaMock/pull/149

**Comment by Maatary** [comment]
Hi Could you explain the 2 step guide. I don't get it. Which files/class needs to be modified ?

**Comment by dvic** [comment]
@Maatary Clone my repo (https://github.com/dvic/ScalaMock) and add all the files from the following folder to your project (copy the sources): `frameworks/scalatest/src/main/scala/org/scalamock/scalatest`

You can copy them to whatever package you want. Then, in your tests, when importing for example `MockFactory`, use the version from your own package and not the one from scalamock.

**Comment by jeremyrsmith** [comment]
I extracted `scalamock-scalatest-support` into a separate module with @fommil's changes:

https://github.com/jeremyrsmith/scalamock-scalatest-support

It's published to Bintray for ease of use. Happy to remove it once @paulbutcher is able to revisit ScalaMock.

**Comment by mach-kernel** [comment]
@jeremyrsmith thank you for your bintray release, it solved my issues! 👍

**Comment by barkhorn** [comment]
looks like this was fixed in v3.3.0, anye disagreeing?
